extensions/libraries/redcore/translation/helper.php

```diff
@@ -609,4 +610,219 @@ public static  function setLanguage($language, $loadLanguage = true)
 
 		return false;
 	}
+
+	/**
+	 * Checks if the current page is a translatable form.
+	 *
+	 * @param   bool  $isAdmin  Informs us whether we are on frontend or backend
+	 *
+	 * @return  void
+	 */
+	public static function isTranslatableForm($isAdmin)
+	{
+		// Current page values
+		$input = JFactory::getApplication()->input;
+		$option = $input->getString('option', '');
+		$view = $input->getString('view', '');
+		$layout = $input->getString('layout', '');
+		$task = $input->getString('task', '');
+
+		$translationTables = self::getInstalledTranslationTables();
+
+		foreach ($translationTables as $tableKey => $translationTable)
+		{
+			if (!isset($translationTable->formLinks))
+			{
+				continue;
+			}
+
+			foreach ($translationTable->formLinks as $formLink)
+			{
+				// Form values
+				$tableAdmin = $formLink['admin'];
+				$tableOption = $formLink['option'];
+				$tableView = $formLink['view'];
+				$tableLayout = $formLink['layout'];
+				$tableID = $formLink['identifier'];
+				$showButton = $formLink['showbutton'];
+				$htmlposition = $formLink['htmlposition'];
+
+				// Check if the form's frontend/backend options matches the current page
+				$tableAdmin = $tableAdmin === 'true' ? true : false;
+
+				if ($isAdmin != $tableAdmin)
+				{
+					continue;
+				}
+
+				// Check whether form values matches the current page
+				if ($option == $tableOption && $view == $tableView && $layout == $tableLayout)
+				{
+					// Get id of item based on the form identifier.
+					$itemID = $input->getInt($tableID, '');
+
+					// If the item doesn't have an ID, tell the user that they have to save the item first.
+					if (empty($itemID))
+					{
+						self::renderTranslationModal(false, false, false, false);
+
+						return;
+					}
+
+					// Check whether there's a relation between the current item and the translation element
+					$db = JFactory::getDbo();
+					$query = $db->getQuery(true)
+					->select($db->qn($translationTable->primaryKeys[0]))
+					->from($db->qn($translationTable->table))
```

Comment:
Yes, this is intentional.
It's an extra check to ensure that the item actually exists in the original table, so that you don't get unnecessary translation buttons and phantom translations.

Take something like redITEM or Aesir for example, where each type's custom fields are saved in a separate table, but are still edited in the same form. If we don't make this check, it'll show translation buttons for every type that has a translation specification, rather than just the relevant ones.